### PR TITLE
Added complete list of common request and response headers. Added mor…

### PR DESCRIPTION
**Pull request by Koc**
…e autocomlete for some of core functions and Symfony classes.

### CODE REVIEW

**Comment by Haehnchen** [comment]
will pipe more PhpIndex filters like `traits`, `interfaces`, `functions`

**Comment by Koc** [comment]
@Haehnchen I've updated for `trait` and `interface` providers

**Comment by Koc** [comment]
@Haehnchen What about name providers in plural and prefix it? Like `class` -> `core.classes`, `http_request_headers` -> `core.http.request_headers`

**Comment by stof** [comment]
IMO, you should avoid changing indentation

**Comment by Haehnchen** [comment]
@Koc currently i am not happy with current naming. your idea looks much better. but there is also another use case which need to be think of. 

``` php
/**
* @var string $foo my nice stuff #ClassInterface foo
* @var string $foo my nice stuff #http_request_headers foo
* @var string $foo my nice stuff #core.http.request_headers bar
*/
function($foo) {}
```

File: symfony/.ide-toolbox.metadata.json [hunk 1]
```diff
@@ -1,55 +1,134 @@
...
+        //            "provider": "class_interface",
+        //            "language": "php",
+        //            "signatures": [
+        //                {
+        //                    "class": "Symfony\\Component\\HttpKernel\\DependencyInjection\\Extension",
+        //                    "method": "addClassesToCompile",
+        //                    "type": "array_value"
+        //                }
+        //            ]
+        //        },
```
Comment (stof) [inline symfony/.ide-toolbox.metadata.json]: comments are not allowed in json

Comment (Koc) [inline symfony/.ide-toolbox.metadata.json]: but it not fails and ok parsed

**Comment by Koc** [comment]
@Haehnchen pleas provide more information about this case. Referencing from phpdocs to providers?

**Comment by Koc** [comment]
Ported file opening modes, thanx to @King2500

**Comment by Koc** [comment]
@Haehnchen as you have added support of parsing providers name with dot, what about renaming core providers? Also does anything block this PR merging?

**Comment by Haehnchen** [comment]
please remove comments. i will then cleanup provider names after merge

**Comment by Koc** [comment]
@Haehnchen done

**Pull request merged**